benchmark/single-source/WordCount.swift

```diff
@@ -0,0 +1,253 @@
+  let baseAlphabet = Set(
+    "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ".unicodeScalars)
+  let someCombiningDiacriticalMarks: Set<Unicode.Scalar> =
+    Set((0x300..<0x310).map { Unicode.Scalar($0)! })
+
+  var alphanumerics: Set<Character> = []
+  for base in baseAlphabet {
+    alphanumerics.insert(Character(base))
+    for mark in someCombiningDiacriticalMarks {
+      var v = String.UnicodeScalarView()
+      v.append(base)
+      v.append(mark)
+      alphanumerics.insert(Character(String(v)))
+    }
+  }
+  alphanumerics.formUnion("0123456789")
+  return alphanumerics
+}()
+
+extension Character {
+  var isAlphanumeric: Bool {
+    return someAlphanumerics.contains(self)
+  }
+}
+
+struct Words: IteratorProtocol, Sequence {
+  let text: String
+  var nextIndex: String.Index
+
+  init(_ text: String) {
+    self.text = text
+    self.nextIndex = text.startIndex
+  }
+
+  func makeIterator() -> Words {
+    return self
+  }
+
+  mutating func next() -> String? {
+    while nextIndex != text.endIndex && !text[nextIndex].isAlphanumeric {
+      text.formIndex(after: &nextIndex)
+    }
+    let start = nextIndex
+    while nextIndex != text.endIndex && text[nextIndex].isAlphanumeric {
+      text.formIndex(after: &nextIndex)
+    }
+    guard start < nextIndex else { return nil }
+    return String(text[start..<nextIndex])
+  }
+}
+
+@inline(never)
+public func run_WordSplitASCII(_ N: Int) {
+  for _ in 1...10*N {
+    let count = Array(Words(identity(asciiText))).count
+    guard count == 280 else { fatalError() }
+  }
+}
+
+@inline(never)
+public func run_WordSplitUTF16(_ N: Int) {
+  for _ in 1...10*N {
+    let count = Array(identity(Words(utf16Text))).count
+    guard count == 280 else { fatalError() }
+  }
+}
+
+let asciiWords = Array(Words(asciiText))
+let utf16Words = Array(Words(utf16Text))
+
+@inline(never)
+public func run_WordCountUniqueASCII(_ N: Int) {
+  for _ in 1...100*N {
+    let words = Set(identity(asciiWords))
+    guard words.count == 168 else { fatalError() }
+    blackHole(words)
+  }
+}
+
+@inline(never)
+public func run_WordCountUniqueUTF16(_ N: Int) {
+  for _ in 1...100*N {
+    let words = Set(identity(utf16Words))
+    guard words.count == 168 else { fatalError() }
+    blackHole(words)
+  }
+}
+
+/// Returns an array of all words in the supplied string, along with their
+/// number of occurances. The array is sorted by decreasing frequency.
+/// (Words are case-sensitive and only support a limited subset of Unicode.)
+@inline(never)
+func histogram<S: Sequence>(for words: S) -> [(String, Int)]
+where S.Element == String {
+  let histogram = Dictionary<String, Int>(
+    words.lazy.map { ($0, 1) },
+    uniquingKeysWith: +)
+  return histogram.sorted { (-$0.1, $0.0) < (-$1.1, $1.0) }
+}
+
+@inline(never)
+public func run_WordCountHistogramASCII(_ N: Int) {
+  for _ in 1...100*N {
+    let words = histogram(for: identity(asciiWords))
+    guard words.count == 168 else { fatalError() }
```

Comment:
Looks like a nice place to reuse `CheckResults` from `TestUtils` that's used elsewhere in the benchmarks.

Reply:
https://github.com/apple/swift/blob/master/benchmark/utils/TestsUtils.swift#L161